### ISSUE TITLE
Properly register middlewares when using NewGroupPath

### ISSUE THREAD

**Issue by teja42** [savsgio/atreugo#59]
Consider this : 

```
	AuthRouter = Server.Router.NewGroupPath("/auth")

	// Authentication middleware
	Server.Router.UseBefore(func(ctx *atreugo.RequestCtx) error {
		token := ctx.Request.Header.Peek("Authorization")
		claims, err := jwt.ECDSACheck(token, jwtPublicKey)
		if err != nil || !claims.Valid(time.Now()) {
			ctx.JSONResponse(map[string]interface{}{
				"err": "You have been logged out. Login to continue",
			}, 401)
			return nil
		}

		ctx.SetUserValue("auth_username", claims.Subject)
		ctx.Next()
		return nil
	})

	UploadRouter = Server.Router.NewGroupPath("/upload")
```

AuthRouter & UploadRouter are exported and the paths are defined in another package. 
So right now there are no paths ( or routes ) defined.
This makes the middleware register first causing it to run before any route.
I expected that the middleware would run if the route is anything other than /auth because of the order in which they were defined. But it appears that NewGroupPath has no effect in order of middleware registration. I don't think it's really helpful this way.

**Comment by savsgio** [comment]
Hi @teja42,

The authentication middleware must be executed in the two group paths (AuthRouter and UploadRouter) due to you are registering on main router not in group router.  So independently, if you declare the auth middleware before or after the group routers or in another sub-package, the execution always be the same, don't mind the declaration order in the code, just the `UseAfter` or `UseBefore`.

Execution chain:

1. Before middleware
2. Before middleware (group)
3. Before filter
4. View
5. After filter
6. After middleware (group)
7. After middleware

Because you want execute the authentication middleware in all paths, group paths, etc, right?

**Comment by teja42** [comment]
Sorry for delayed response.

I don't want the middleware to run for paths that start with `/auth`
I want the middleware to run before any path that doesn't start with `/auth`

I'm not sure this is possible right now or I may me wrong. If this is possible can you give me a simple example based on my original example

**Comment by teja42** [comment]
Sorry for the delayed response

I DONT want the middleware to run for paths start with /auth
I WANT the middleware to run for paths that don't start with /auth

What I ended up doing was I registered the middleware first and if the path starts with `/auth` then  
I called `ctx.Next()`. If not I performed the authentication and did accordingly. This approach works. Suggest me If there is a better one.

**Comment by savsgio** [comment]
At now it's the best solution

But i'm thinking in a new method to simplify that.

Soon, it could be available.

Thanks!

**Comment by savsgio** [comment]
Hi @teja42,

I've been released a new version [v10.3.0](https://github.com/savsgio/atreugo/releases/tag/v10.3.0)

I added support for skip middlewares in groups and views 🎉

**Comment by teja42** [comment]
Awesome. Thanks for the feature.
